src/presign/participant.rs

```diff
@@ -399,7 +568,21 @@ impl PresignParticipant {
             ));
         }
 
-        Ok(ret_messages)
+        // Additionally, handle any round 3 messages which may have been received too
+        // early
+        let mut presign_record = None;
+        let retrieved_messages = self.fetch_messages(
+            MessageType::Presign(PresignMessageType::RoundThree),
```

Comment:
I notice this pattern `MessageType::Presign(PresignMessageType::RoundThree)` throughout this file. Would it make sense to create some helper functions like so:
```rust
impl MessageType {
    fn round_three_presign() -> MessageType {
        MessageType::Presign(PresignMessageType::RoundThree)
    }
}
```
And then you can just call `MessageType::round_three_presign()` instead. (Same for round 1 and two).

Reply:
Maybe this would be a little cleaner? But it would apply to all the subprotocols so is maybe out of scope for now. Maybe we can discuss this when going over message processing tomorrow